### PR TITLE
[Frontend][Feature] Add jamba tool parser

### PR DESCRIPTION
**Pull request by tomeras91**
Add `JambaToolParser` to support tool calling for [ai21labs/AI21-Jamba-1.5-Mini](https://huggingface.co/ai21labs/AI21-Jamba-1.5-Mini) and [ai21labs/AI21-Jamba-1.5-Large](https://huggingface.co/ai21labs/AI21-Jamba-1.5-Large)

### CODE REVIEW

**Comment by DarkLight1337** [comment]
@K-Mistele do you have time to review this?

**Comment by K-Mistele** [comment]
> @K-Mistele do you have time to review this?

Yes, happy to take a look!

File: vllm/entrypoints/openai/tool_parsers/jamba_tool_parser.py
Comment (mgoin) [inline vllm/entrypoints/openai/tool_parsers/jamba_tool_parser.py:85]: Is this TODO done?

**Review by mgoin — COMMENTED** [review]
This seems reasonable to me but I will wait on Kyle to review

File: vllm/entrypoints/openai/tool_parsers/jamba_tool_parser.py
Comment (tomeras91) [reply vllm/entrypoints/openai/tool_parsers/jamba_tool_parser.py:85]: Yes. Removed it now.
Thanks for catching this

**Comment by tomeras91** [comment]
> > @K-Mistele do you have time to review this?
> 
> Yes, happy to take a look!

Ping? Anything I can do to help this getting merged?

**Review by K-Mistele — CHANGES_REQUESTED** [review]
Overall looks good! just a couple thoughts:

1. Right now, tool use tests are already standardized across models in `tests/tool_use`, you can add a config for Jamba in [`vllm/tests/tool_use/utils.py`](https://github.com/vllm-project/vllm/blob/main/tests/tool_use/utils.py) so that the same tests that are run on other models are run on it. This is preferred since it keeps tool testing standardized to make sure that all models pass the same tests, rather than defining custom tests for each model
2. Please update the docs at `docs/source/serving/openai_compatible_server.md` with the relevant information for jamba - you'll see that Llama 3.1, Hermes, Mistral, and InternLM already have entries here, so you should be able to use the same template.

**Comment by tomeras91** [comment]
@K-Mistele
1. RE tests - that was my initial thought as well. Problem is even [Jamba-1.5-Mini](https://huggingface.co/ai21labs/AI21-Jamba-1.5-Mini) is a pretty large model with ~52B params. It will take a long time to download every time the unittest is run. Moreover, even with quantization it needs an 80GB GPU, and AFAIU tests currently run on L4 GPUs.. so we'll have issues with memory as well. The [Jamba-tiny-dev](https://huggingface.co/ai21labs/Jamba-tiny-dev) model used for other Jamba tests is a pretrained model and doesn't support tool calling. 
Other than the model size issues, the end2end behavior of tool calling is already tested in tool calling tests of other models. Also, that end2end logic hasn't changed in this PR. The only logic this PR adds is the JambaToolParser logic. And conveniently enough (AKA good code design ny you guys), it can be thoroughly tested without initializing a model, as I did in tests/tool_use/test_jamba_tool_parser.py. All it needs is the tokenizer and a raw string treated as the model output.

2. RE docs - sure! I'll update them soon

**Comment by K-Mistele** [comment]
> @K-Mistele
> 
> 1. RE tests - that was my initial thought as well. Problem is even [Jamba-1.5-Mini](https://huggingface.co/ai21labs/AI21-Jamba-1.5-Mini) is a pretty large model with ~52B params. It will take a long time to download every time the unittest is run. Moreover, even with quantization it needs an 80GB GPU, and AFAIU tests currently run on L4 GPUs.. so we'll have issues with memory as well. The [Jamba-tiny-dev](https://huggingface.co/ai21labs/Jamba-tiny-dev) model used for other Jamba tests is a pretrained model and doesn't support tool calling.
>    Other than the model size issues, the end2end behavior of tool calling is already tested in tool calling tests of other models. Also, that end2end logic hasn't changed in this PR. The only logic this PR adds is the JambaToolParser logic. And conveniently enough (AKA good code design by you guys), it can be thoroughly tested without initializing a model, as I did in tests/tool_use/test_jamba_tool_parser.py. All it needs is the tokenizer and a raw string treated as the model output.
> 2. RE docs - sure! I'll update them soon

Sounds good! I didn't realize the model was so big. I will defer to @mgoin @DarkLight1337 then.

**Review by DarkLight1337 — APPROVED** [review]
LGTM, thanks for your patience!

**Pull request merged**
